File: backend/src/main/java/com/str/backend/model/RedeemedReward.java
package com.str.backend.model;

import jakarta.persistence.*;

import java.time.LocalDateTime;

@Entity
public class RedeemedReward {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String qrCode;

    private LocalDateTime redeemedAt;

    private boolean used = false;

    private LocalDateTime usedAt;

    @ManyToOne
    private User user;

    @ManyToOne
    private Reward reward;

    public RedeemedReward() {
    }

    public Long getId() {
        return id;
    }

    public String getQrCode() {
        return qrCode;
    }

    public void setQrCode(String qrCode) {
        this.qrCode = qrCode;
    }

    public LocalDateTime getRedeemedAt() {
        return redeemedAt;
    }

    public void setRedeemedAt(LocalDateTime redeemedAt) {
        this.redeemedAt = redeemedAt;
    }

    public boolean isUsed() {
        return used;
    }

    public void setUsed(boolean used) {
        this.used = used;
    }

    public LocalDateTime getUsedAt() {
        return usedAt;
    }

    public void setUsedAt(LocalDateTime usedAt) {
        this.usedAt = usedAt;
    }

    public User getUser() {
        return user;
    }

    public void setUser(User user) {
        this.user = user;
    }

    public Reward getReward() {
        return reward;
    }

    public void setReward(Reward reward) {
        this.reward = reward;
    }
}

File: frontend/lib/models/redeemed_reward.dart
class RedeemedReward {

  final int id;
  final String rewardName;
  final String qrCode;
  final String redeemedAt;
  final bool used;

  RedeemedReward({
    required this.id,
    required this.rewardName,
    required this.qrCode,
    required this.redeemedAt,
    required this.used,
  });

  factory RedeemedReward.fromJson(
    Map<String, dynamic> json,
  ) {

    return RedeemedReward(
      id: json['id'],
      rewardName: json['reward']['name'],
      qrCode: json['qrCode'],
      redeemedAt: json['redeemedAt'],
      used: json['used'],
    );
  }
}